File: atcoder/ABS/python3/j.py

s = input()
s = s[::-1]
strs = ['dream','dreamer','erase','eraser']
strs = [t[::-1] for t in strs]

while s:
    ans = True
    for i in range(4):
            if(s[0:len(strs[i])] == strs[i]):
                s = s[len(strs[i]):]
                ans = False
    if (ans) :
        break

if s :
    print("NO")
else :
    print("YES")
        

File: atcoder/ABS/php/j.php
<?php

$s = trim(fgets(STDIN));

$words = array("erase","dream","eraser","dreamer");

while($s){
  for($i=0;$i<5;$i++){
    if($i==4){
      print "NO\n";
      return;
    }else{
      $len = strlen($words[$i]);
      $subs =substr($s,-($len));
      //print $subs." ".$words[$i]."\n";
      if(strcmp($subs,$words[$i])==0){

        $s=substr($s,0,-$len);
        break;
      }
    }
  }
}

print "YES\n";

?>

